cmd/influx/v1_dbrp.go

```diff
@@ -0,0 +1,386 @@
+		mapping:     dbrp,
+	})
+}
+
+type v1DBRPPrintOpt struct {
+	jsonOut     bool
+	hideHeaders bool
+	mapping     *influxdb.DBRPMappingV2
+	mappings    []*influxdb.DBRPMappingV2
+}
+
+func v1WriteDBRPs(w io.Writer, printOpts v1DBRPPrintOpt) error {
+	if printOpts.jsonOut {
+		var v interface{} = printOpts.mappings
+		if printOpts.mappings == nil {
+			v = printOpts.mappings
+		}
+		return writeJSON(w, v)
+	}
+
+	tabW := internal.NewTabWriter(w)
+	defer tabW.Flush()
+
+	tabW.HideHeaders(printOpts.hideHeaders)
+
+	headers := []string{
+		"ID",
+		"Database",
+		"Bucket ID",
+		"Retention Policy",
+		"Default",
+		"Organization ID",
+	}
+
+	tabW.WriteHeaders(headers...)
+
+	if printOpts.mappings == nil {
+		printOpts.mappings = append(printOpts.mappings, printOpts.mapping)
+	}
+
+	for _, t := range printOpts.mappings {
+		m := map[string]interface{}{
+			"ID":               t.ID.String(),
+			"Database":         t.Database,
+			"Retention Policy": t.RetentionPolicy,
+			"Default":          t.Default,
+			"Organization ID":  t.OrganizationID,
+			"Bucket ID":        t.BucketID,
+		}
+		tabW.Write(m)
+	}
+
+	return nil
+}
+
+var v1DBRPDeleteFlags struct {
+	ID  string       // Specifies the mapping ID to filter on
+	Org organization // required  // Specifies the organization ID to filter on
+}
+
+func v1DBRPDeleteCmd(f *globalFlags, opt genericCLIOpts) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete DBRP",
+		RunE:  checkSetupRunEMiddleware(&flags)(v1DBRPDeleteF),
+	}
+
+	f.registerFlags(opt.viper, cmd)
+	registerPrintOptions(opt.viper, cmd, &v1DBRPCRUDFlags.hideHeaders, &v1DBRPCRUDFlags.json)
+	v1DBRPCreateFlags.Org.register(opt.viper, cmd, false)
+	cmd.Flags().StringVarP(&v1DBRPDeleteFlags.ID, "id", "i", "", "The dbrp ID (required)")
+	_ = cmd.MarkFlagRequired("id")
+	return cmd
+}
+
+func v1DBRPDeleteF(cmd *cobra.Command, args []string) error {
+	if err := v1DBRPCreateFlags.Org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+	orgSvc, err := newOrganizationService()
+	if err != nil {
+		return err
+	}
+
+	orgID, err := v1DBRPDeleteFlags.Org.getID(orgSvc)
+	if err != nil {
+		return err
+	}
+
+	s, err := newV1DBRPService()
+	if err != nil {
+		return err
+	}
+
+	id, err := influxdb.IDFromString(v1DBRPDeleteFlags.ID)
+	if err != nil {
+		return err
+	}
+
+	dbrpToDelete, err := s.FindByID(context.Background(), orgID, *id)
+	if err != nil {
+		return err
+	}
+
+	if err := s.Delete(context.Background(), orgID, *id); err != nil {
+		return err
+	}
+	return v1WriteDBRPs(cmd.OutOrStdout(), v1DBRPPrintOpt{
+		jsonOut:     v1DBRPCRUDFlags.json,
+		hideHeaders: v1DBRPCRUDFlags.hideHeaders,
+		mapping:     dbrpToDelete,
+	})
+}
+
+var v1DBRPUpdateFlags struct {
+	ID      string       // Specifies the mapping ID to filter on
+	Org     organization // required  // Specifies the organization ID to filter on
+	DB      string       //InfluxDB v1 database
+	Default *bool
+	RP      string // InfluxDB v1 retention policy
+}
```

Comment:
Should we also have BucketID on this one?

Reply:
The Server ignores everything except RP and Default.  Thanks, though I had forgotten to remove DB from the object (but I removed it from the flags